PWGHF/Core/HFMLResponse.h

```diff
@@ -0,0 +1,212 @@
+      mNetworks[counterModel].initModel(path, enableOptimizations, threads);
+      ++counterModel;
+    }
+  }
+
+  // Initialize OnnxModels when paths already defined
+  /// \param paths is a vector of onnx model paths
+  /// \param enableOptimizations is a switch no enable optimizations
+  /// \param threads
+  void init(bool enableOptimizations = false, int threads = 0)
+  {
+    return init(mPaths, enableOptimizations, threads);
+  }
+
+  /// Get vector with model predictions
+  /// \param input a vector containing the values of features used in BDT model
+  /// \param nModel is the model index
+  /// \return model prediction for each class and each model
+  template <typename T1, typename T2>
+  std::vector<T> getModelOutput(T1& input, const T2& nModel)
+  {
+    T* outputPtr = mNetworks[nModel].evalModel(input);
+    std::vector<T> output(outputPtr, outputPtr + mNClasses);
+    return output;
+  }
+
+  /// ML selections
+  /// \param input is the input features
+  /// \param nModel is the model index
+  /// \return boolean telling if model predictions pass the cuts
+  template <typename T1, typename T2>
+  bool isSelectedML(T1& input, const T2& nModel)
+  {
+    auto output = getModelOutput(input, nModel);
+    uint8_t iClass{0};
+    for (const auto& outputValue : output) {
+      uint8_t dir = mCutDir.at(iClass);
+      if (dir != hf_cuts_ml::CutDirection::CutNot) {
+        if (dir == hf_cuts_ml::CutDirection::CutGreater && outputValue > mCuts.get(nModel, iClass)) {
+          return false;
+        } else if (dir == hf_cuts_ml::CutDirection::CutSmaller && outputValue < mCuts.get(nModel, iClass)) {
+          return false;
+        }
+      }
+      ++iClass;
+    }
+    return true;
+  }
+
+  /// ML selections
+  /// \param input is the input features
+  /// \param nModel is the model index
+  /// \param output is a container to be filled with model output
+  /// \return boolean telling if model predictions pass the cuts
+  template <typename T1, typename T2>
+  bool isSelectedML(T1& input, const T2& nModel, std::vector<T>& output)
+  {
+    output = getModelOutput(input, nModel);
+    uint8_t iClass{0};
+    for (const auto& outputValue : output) {
+      uint8_t dir = mCutDir.at(iClass);
+      if (dir != hf_cuts_ml::CutDirection::CutNot) {
+        if (dir == hf_cuts_ml::CutDirection::CutGreater && outputValue > mCuts.get(nModel, iClass)) {
+          return false;
+        } else if (dir == hf_cuts_ml::CutDirection::CutSmaller && outputValue < mCuts.get(nModel, iClass)) {
+          return false;
+        }
+      }
+      ++iClass;
+    }
+    return true;
+  }
+
+ private:
+  std::vector<o2::ml::OnnxModel> mNetworks;       // OnnxModel objects, one for each bin
+  uint8_t mNModels = 1;                           // number of bins
+  uint8_t mNClasses = 3;                          // number of model classes
+  std::vector<double> mBinsLimits = {};           // bin limits of the variable (e.g. pT) used to select which model to use
```

Comment:
This variable is not used at the moment since the index to the model is provided by the "user". Maybe the selection of which model to use could be integrated in this class?

Reply:
You mean feeding `candidate.pt()` to `isSelectedML` instead of the `pTBin`?

Reply:
Yes

Reply:
Done :)